app/controllers/storage_location_repository_rows_controller.rb

```diff
@@ -19,22 +19,46 @@ def index
 
   def create
     ActiveRecord::Base.transaction do
-      @storage_location_repository_row = StorageLocationRepositoryRow.new(
-        repository_row: @repository_row,
-        storage_location: @storage_location,
-        metadata: storage_location_repository_row_params[:metadata] || {},
-        created_by: current_user
-      )
-
-      @storage_location_repository_row.with_lock do
-        if @storage_location_repository_row.save
-          log_activity(:storage_location_repository_row_created)
-          render json: @storage_location_repository_row,
-                 serializer: Lists::StorageLocationRepositoryRowSerializer
-        else
-          render json: { errors: @storage_location_repository_row.errors.full_messages }, status: :unprocessable_entity
+      new_assignments = []
+
+      if @storage_location.with_grid?
+        params[:positions].each do |position|
+          if position.dig(2, :occupied)
+            occupied_row = @storage_location.storage_location_repository_rows.find_by(id: position.dig(2, :id))
+            raise ActiveRecord::RecordInvalid, occupied_row unless discard_storage_location_repository_rows(occupied_row)
+          end
+          row = StorageLocationRepositoryRow.new(
+            repository_row: @repository_row,
+            storage_location: @storage_location,
+            metadata: { position: position[0..1] },
+            created_by: current_user
+          )
+          row.with_lock do
+            row.save!
+            new_assignments << row
+          end
+        end
+      else
+        row = StorageLocationRepositoryRow.new(
+          repository_row: @repository_row,
+          storage_location: @storage_location,
+          created_by: current_user
+        )
+        row.with_lock do
+          row.save!
+          new_assignments << row
```

Comment:
We should just call it what it is, it doesn't matter if it's longer:
`storage_location_repository_row = StorageLocationRepositoryRow.new(`
and
`storage_location_repository_rows << storage_location_repository_row`

The same in other places in this controller.